quartz/depgraph.ts

```diff
@@ -0,0 +1,123 @@
+import Graph from "graphology"
```

Comment:
do we need to depend on `graphology` here? it seems like we already implement most of the graph algorithms ourself so I'm not sure what we're getting with the dependency

Reply:
I could take a stab at it. Not too confident I'll handle all the edge cases in a graph data structure though -- wdyt about leaving it as is for now and potentially removing in a future PR?

Reply:
Not as tricky as I thought, removed the dep